packages/qwik/src/core/api.md

```diff
@@ -623,7 +623,7 @@ export type PropFnInterface<ARGS extends any[], RET> = {
 // @public (undocumented)
 export type PropFunction<T extends Function = (...args: any) => any> = T extends (...args: infer ARGS) => infer RET ? PropFnInterface<ARGS, Awaited<RET>> : never;
 
-// @public (undocumented)
+// @public @deprecated (undocumented)

```

Comment:
I wonder if we should add this, the way I understand it, deprecation causes eslint to fail in some codebases, and it's a bit annoying to remove all the old uses.

That would be a semi-breaking change. We can keep PropFunctionProps in the codebase indefinitely and just say you shouldn't use it?

But if there's no eslint rule that breaks, then I'm ok with deprecation.

Reply:
PropFunctionProps was simply inferring "PropFnInterface" to props. It was never documented so I doubt a lot of people were using this type extensively, especially since having to create custom events and use `PropFunction` is pretty rare imo.

I'd argue it's better to explicitly notice the developers who are using PropFunctionProps (which shouldn't be a lot of devs) that it's deprecated. Better know it soon than later no? I think that can even be marked in a patch release since it wasn't a documented API.